### PR TITLE
Feature/rhp 169 dont show all events in events list

### PR DESCRIPTION
**Pull request by johngreedjr**

### CODE REVIEW

File: app/api/eventsController.js [hunk 1]
```diff
@@ -35,114 +16,56 @@
...
-          } else {
-            res.send(416, {
-              message: 'An event already exists.',
-              code: 'EVENT_ALREADY_EXISTS'
-            })
-          }
-        })
-      }
+      eventsService.persistEvent(req.body).then((ev) => {
+        res.send(ev);
```
Comment (johngreedjr) [inline app/api/eventsController.js]: add a return

File: app/api/eventsController.js [hunk 2]
```diff
@@ -35,114 +16,56 @@
...
 
   api.get('/date', function(req, res) {
-    var startDate = moment(req.query.startDate).startOf('day').format();
-    var endDate;
-    if (!req.query.endDate) {
-      endDate = req.query.startDate;
-    }
-    endDate = moment(endDate).endOf('day').format();
+    eventsService.getByDate(req.query.startDate, req.query.endDate).then((events) => {
+      res.send(events);
```
Comment (brenthaertlein) [inline app/api/eventsController.js]: Use `return` consistently.

File: app/api/eventsService.js [hunk 3]
```diff
@@ -2,15 +2,57 @@
...
+      path: 'venue',
+      select: 'name day time numberOfGames'
+    }, {
+      path: 'td',
+      select: '-statusId'
+    }, {
+      path: 'games'
+    }];
+
     function createEvent(event, cb) {
```
Comment (brenthaertlein) [inline app/api/eventsService.js]: Refactor this to return a `Promise` and update all references.

File: app/api/eventsService.js [hunk 4]
```diff
@@ -26,15 +68,24 @@
...
+            if (err) {
+              console.error(err);
+              return reject(err);
+            }
+            return resolve(ev);
+          })
+      })
+    }
 
     function getEvents(cb) {
```
Comment (brenthaertlein) [inline app/api/eventsService.js]: Refactor this to return a `Promise` and update all references.

File: app/api/eventsService.js [hunk 5]
```diff
@@ -51,6 +102,21 @@
...
+            if (err) {
+              console.error(err);
+              return reject(err);
+            }
+            return resolve(event);
+          })
+      })
+    }
+
     function getPastEvents(cb) {
```
Comment (brenthaertlein) [inline app/api/eventsService.js]: Refactor this to return a `Promise` and update all references.

**Review by brenthaertlein — CHANGES_REQUESTED** [review]

**Review by brenthaertlein — APPROVED** [review]

**Pull request merged**